app/controllers/Application.scala

```diff
@@ -154,7 +154,11 @@ object Application extends Controller {
   def npmPackageVersions(packageName: String) = Action.async {
     val packageVersionsFuture = Cache.getAs[Seq[String]](s"npm-versions-$packageName").fold {
       npm.info(packageName).map { json =>
-        val versions = (json \ "versions" \\ "version").map(_.as[String])
+        val versions = (json \ "versions" \\ "version")
+          .map(_.as[String])
+          .map(WebJarVersion(_))
+          .sorted(WebJarVersion.WebJarVersionOrdering.reverse)
+          .map(_.number)
```

Comment:
Just realized what I said didn't make sense, didn't make sense. I should stop trying to code hastily around work.

I had tried
```scala
import WebJar._
val versions = (json \ "versions" \\ "version").map(_.as[WebJarVersion]).sorted
```

which clearly wouldn't work, because the numFiles field isn't optional as I had mistakenly attributed, but just contains a default value.

As for refactoring, what did you have in mind?